app/models/internal/impl/DefaultReport.java

```diff
@@ -0,0 +1,167 @@
+        _source = builder._source;
+        _groups = builder._groups;
+    }
+
+    @Override
+    public UUID getId() {
+        return _id;
+    }
+
+    @Override
+    public String getName() {
+        return _name;
+    }
+
+    @Override
+    public Schedule getSchedule() {
+        return _schedule;
+    }
+
+    @Override
+    public ReportSource getSource() {
+        return _source;
+    }
+
+    @Override
+    public Collection<RecipientGroup> getRecipientGroups() {
+        return Collections.unmodifiableCollection(_groups);
+    }
+
+    @Override
+    public Optional<Instant> getLastRun() {
+        return Optional.empty();
+    }
+
+    @Override
+    public CompletionStage<Report.Result> execute(final ActorRef scheduler, final Instant scheduled) {
+        return CompletableFuture.completedFuture(null);
+    }
+
+    private final UUID _id;
+    private final String _name;
+    private final Schedule _schedule;
+    private final ReportSource _source;
+    private final Collection<RecipientGroup> _groups;
+
+    /**
+     * Builder implementation that constructs {@code DefaultReport}.
+     */
+    public static final class Builder extends OvalBuilder<DefaultReport> {
+        /**
+         * Public Constructor.
+         */
+        public Builder() {
+            super(DefaultReport::new);
+            _schedule = NeverSchedule.getInstance();
+            _groups = Collections.emptySet();
+        }
+
+        /**
+         * Set the report id. Required. Cannot be null.
+         * @param id The report id.
+         * @return This instance of {@code Builder}.
+         */
+        public Builder setId(final UUID id) {
+            _id = id;
+            return this;
+        }
+
+        /**
+         * Set the report name. Required. Cannot be null or empty.
+         * @param name The report name.
+         * @return This instance of {@code Builder}.
+         */
+        public Builder setName(final String name) {
+            _name = name;
+            return this;
+        }
+
+        /**
+         * Set the report schedule. Required. Cannot be null.
+         * @param schedule The report schedule.
+         * @return This instance of {@code Builder}.
+         */
+        public Builder setSchedule(final Schedule schedule) {
+            _schedule = schedule;
+            return this;
+        }
+
+        /**
+         * Set the report source. Required. Cannot be null.
+         * @param source The report source.
+         * @return This instance of {@code Builder}.
+         */
+        public Builder setReportSource(final ReportSource source) {
+            _source = source;
+            return this;
+        }
+
+        /**
+         * Set the report recipients. Required. Cannot be null.
+         * @param groups The report recipient groups.
+         * @return This instance of {@code Builder}.
+         */
+        public Builder setRecipientGroups(final Collection<RecipientGroup> groups) {
```

Comment:
The problem with accepting Collection like this is you don't know whether you need to create a copy; otherwise your caller can mutate your internal model object thus breaking your immutability.

Instead, we generally prefer to accept Guava's `Immutable*` to avoid the copy. If the caller doesn't have one then they are forced to copy, so the cost is never any worse, but the contract is clearer, immutability is preserved and when the caller does have an `Immutable*` it's zero-copy.